openedx/features/philu_courseware/views.py

```diff
@@ -51,10 +51,9 @@ def record_and_fetch_competency_assessment(request):
         score:Integer, 1
     :return: JSON
     """
-    competency_record = request.POST.dict()
-    competency_record['user'] = request.user.id
+    competency_record = request.data
```

Comment:
How about changing it to following and use `chapter_id` and `competency_assessment_records` at relevant places.
```
data = request.data
chapter_id = data.get('chapter_id', '')
competency_assessment_records = data.get('competency_assessment_records', [])
```

As I requested a change above to remove `ListCompetencyAssessmentRecordSerializer`, So here we pass `competency_assessment_records` in data and `many=True` and `context=dict(request=request)`

Reply:
I am thinking that we can also get `chapter_id` in url instead of data parameters, Just like we have URL for `competency_assessments_score_view` and in case of coursewares and that would make more sense that we need to save and fetch assessment record in this chapter. 
WDYT?

Reply:
@zeshanarifios to avoid `data.get('chapter_id', '')` I have introduced a new serializer between. I feel it more clean and sophisticated way to achieve the functionality.

Reply:
If you see this line in your PR 
```
competency_assessment_score = get_competency_assessments_score(request.user, competency_record['chapter_id'])
```

You are still using `competency_record['chapter_id']`. I just updated it to use `get` instead of using subscript directly.
I have also suggested another way of avoiding `data.get('chapter_id', '')` to include it in url. We can achieve it in that way if possible in the scope of this ticket.  

IMO It is good to remove the unnecessary Serializer.

Reply:
@zeshanarifios at that point it is validated, as it must be provided in the payload to pass validation, it will be there what I can do is to change from `competency_record` to `serializer.validated_data`. What is your say?

Reply:
It would be great if we can have a call to discuss.